### ISSUE TITLE
shininess is not working in WGLMakie

### ISSUE THREAD

**Issue by gdkrmr** [MakieOrg/Makie.jl#4060]
The following code produces a Javascript error in WGLMakie:

```julia
julia> using WGLMakie

julia> using FileIO

julia> brain = load(assetpath("brain.stl"));

julia> mesh(
           brain,
           color = [tri[1][2] for tri in brain for i in 1:3],
           colormap = Reverse(:Spectral),
           shininess = 64.0
       );

julia> mesh(
           brain,
           color = [tri[1][2] for tri in brain for i in 1:3],
           colormap = Reverse(:Spectral),
           shininess = 64.0
       )
┌ Warning: Error in Javascript: THREE.WebGLProgram: Shader Error No error - VALIDATE_STATUS false
│ 
│ Program Info Log:
│ Must have a compiled vertex shader attached:
│ SHADER_INFO_LOG:
│ ERROR: 0:43: 'double' : Illegal use of reserved word
│ ERROR: 0:43: 'double' : syntax error
│ VERTEX
│ 
│ ERROR: 0:43: 'double' : Illegal use of reserved word
│ ERROR: 0:43: 'double' : syntax error
│ 
│   38: vec3 get_ambient(){return ambient;}
│   39: uniform mat3 normalmatrix;
│   40: mat3 get_normalmatrix(){return normalmatrix;}
│   41: uniform vec4 nan_color;
│   42: vec4 get_nan_color(){return nan_color;}
│ > 43: uniform double shininess;
│   44: double get_shininess(){return shininess;}
...
```

**Comment by SimonDanisch** [comment]
Looks like we don't convert to Float32 automatically... For a workaround you could try `64f0`

**Comment by gdkrmr** [comment]
that works, thanks!

**Comment by abmnmj** [comment]
Testing with the solution suggested in [this discourse thread on type issues with `shininess`](https://discourse.julialang.org/t/using-float64-with-makie-attribute-shininess-gives-error/130293), that is, adding

```Julia
convert_attribute(value, ::key"shininess") = Float32(value)
```

to `conversions.jl` (for example where `convert_attribute(value, ::key"backlight") = Float32(value)` is defined) seems to solve also this issue (tried with the code above and it produces the desired "brain plot" in the browser). So, it would seem that it's related to the same conversion problem.

In the Discourse thread it was suggested that I could make a PR with the above solution, but it seems I don't have the skill set for achieving this. Also, I could not find any example tests that I could base new tests on. 

So, unfortunately I simply add to the present issue to have the (potential) solution documented.

**Comment by asinghvi17** [comment]
That's a temporary patch though.  The real solution is that the shininess values should just have el32convert called on them in float32 backends?

**Comment by ffreyer** [comment]
No, it should just have a convert_attribute. The lighting math combines normalized normals, normalized directions and lighting attributes, so everything revolves around a 0..1 value range. There is no benefit in having extra precision or range there. 

> In the Discourse thread it was suggested that I could make a PR with the above solution, but it seems I don't have the skill set for achieving this. Also, I could not find any example tests that I could base new tests on.

It's fine if you just make a pr with the changes and ask for help with tests, changelog, etc.

I would add another row to this test and try all the lighting attributes with Float64 types there. I would also try to modify the attributes so it looks signifcantly different from the defaults and the case above, so we can verify that it isn't skipping anything
https://github.com/MakieOrg/Makie.jl/blob/eb1ad5b7dfef3b93106760e2b5a1ca59495c0dc9/ReferenceTests/src/tests/examples3d.jl#L150-L162

**Comment by ffreyer** [comment]
The conversion exists now so this should be fixed
